src/test/java/teammates/ui/webapi/GetCourseActionTest.java

```diff
@@ -23,6 +43,84 @@ protected String getRequestMethod() {
         return GET;
     }
 
+    @Test
+    public void testExecute_success() {
+        Course course = generateTypicalCourse();
+
+        GetCourseAction action = generateGetCourseAction();
+        when(action.logic.getCourse(course.getId())).thenReturn(null);
+        when(action.sqlLogic.getCourse(course.getId())).thenReturn(course);
+
+        JsonResult response = action.execute();
+
+        verify(action.logic, times(1)).getCourse(course.getId());
+        verify(action.sqlLogic, times(1)).getCourse(course.getId());
+
+        CourseData courseData = (CourseData) response.getOutput();
+
+        assertEquals(course.getId(), courseData.getCourseId());
+        assertEquals(course.getName(), courseData.getCourseName());
+        assertEquals(course.getTimeZone(), courseData.getTimeZone());
+    }
+
+    private Course generateTypicalCourse() {
+        Course c = new Course("test-courseId", "test-courseName", "test-courseTimeZone", "test-courseInstitute");
+        c.setCreatedAt(Instant.parse("2023-01-01T00:00:00Z"));
+        c.setUpdatedAt(Instant.parse("2023-01-01T00:00:00Z"));
+
+        return c;
+    }
+
+    private Instructor generateTypicalCoOwnerInstructor() {
+        Course course = generateTypicalCourse();
+        InstructorPermissionRole instructorPermissionRole = InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER;
+        InstructorPrivileges instructorPrivileges = new InstructorPrivileges(instructorPermissionRole.getRoleName());
+
+        return new Instructor(
+                course, "test-instructorName", "test@test.com", true,
+                "test-instructorDisplayName", instructorPermissionRole, instructorPrivileges);
+    }
+
+    private GetCourseAction generateGetCourseAction() {
```

Comment:
Just wonder do we need this method? Can we use the `getAction` method?

Reply:
The way I implemented this part was wrong, I did it this way as I was trying to mock all the other classes used in logic layer (like EmailSender, etc) without affecting the other UI layer tests which was unnecessary.

However, since the old UI layer tests will be deleted in future I just left it in.